strapi/strapi
# always escape RegExp when building query in mongoose connector

Fix strapi/strapi#10348



### What does it do?

In a couple places the mongoose connector doesn't escape regexps before passing them to MongoDB

### Why is it needed?

See #10348. In short, making a request to `http://localhost:1337/restaurants?name_contains=lorem\` throws an error because of the trailing backslash.

### How to test it?

Follow the [quickstart](https://strapi.io/documentation/developer-docs/latest/getting-started/quick-start.html#_5-add-content-to-restaurant-content-type) using MongoDB and make a request to `http://localhost:1337/restaurants?name_contains=lorem\`. Without this PR it will throw an error. With this PR, it will return an empty array as expected.

### Related issue(s)/PR(s)

#10348
